### PR TITLE
chore: sort rust derive traits

### PR DESCRIPTION
**Pull request by maksymar**
This PR sorts Rust `#[derive(...)]` traits to maintain a consistent style.

The idea comes from a (now closed) feature request to the `rust-fmt` style team for adding default sorting, [link](https://github.com/rust-lang/style-team/issues/154). 
The sorting follows the 'canonical' order, where `std` traits come first, and the rest follow in alphabetical order. 
It does not modify autogenerated rust code from protobuf files.

The changes were made automatically using the `keepsorted` CLI app, [repo](https://github.com/maksym-arutyunyan/keepsorted), with the `rust_derive_canonical` experimental feature.

### CODE REVIEW

**Review by andrewbattat — APPROVED** [review]

**Review by pakhomov-dfinity — APPROVED** [review]

**Review by Sawchord — APPROVED** [review]

**Review by dsarlis — APPROVED** [review]

**Review by luc-blaeser — APPROVED** [review]

**Review by mbjorkqvist — APPROVED** [review]

**Review by basvandijk — APPROVED** [review]

**Review by mraszyk — APPROVED** [review]

**Review by sesi200 — APPROVED** [review]

File: rs/cross-chain/proposal-cli/src/main.rs [hunk 1]
```diff
@@ -18,8 +18,7 @@ use std::io::Write;
 use std::path::{Path, PathBuf};
 
 /// A fictional versioning CLI
-#[derive(Debug, Parser)] // requires `derive` feature
-#[command(about = "CLI to make canister upgrade proposals", long_about = None)]
```
Comment (gregorydemay) [inline rs/cross-chain/proposal-cli/src/main.rs:22]: This line seems to have been removed, which I think is a mistake.

Comment (maksymar) [reply rs/cross-chain/proposal-cli/src/main.rs:22]: Oh wow! great catch! thank you @gregorydemay !

**Review by blind-oracle — APPROVED** [review]

**Review by DSharifi — APPROVED** [review]

File: rs/cross-chain/proposal-cli/src/main.rs (hunk 1 above)
Comment (maksymar) [reply rs/cross-chain/proposal-cli/src/main.rs]: FYI: I have reviewed all 807 files and didn't find any other cases like that.

**Review by gregorydemay — APPROVED** [review]
Thanks a lot @maksymar for pushing this!

**Review by max-dfinity — APPROVED** [review]

**Pull request merged**